lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -36,6 +36,65 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $columnsByTable = $this->getAssetRecordsByTable($this->_platform->getListColumnsSQL($currentDatabase, null));
+
+        // Get all foreign keys definitions in one database call.
+        $foreignKeysByTable = $this->getAssetRecordsByTable($this->_platform->getListForeignKeysSQL($currentDatabase, null));
+
+        // Get all indexes definitions in one database call.
+        $indexesByTable = $this->getAssetRecordsByTable($this->_platform->getListIndexesSQL($currentDatabase, null));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = trim($tableName, '"');
+
+            $columns = $this->_getPortableTableColumnList($tableName, null, $columnsByTable[$unquotedTableName]);
+
+            $foreignKeys = [];
+            if (isset($foreignKeysByTable[$unquotedTableName])) {
+                $foreignKeys = $this->_getPortableTableForeignKeysList($foreignKeysByTable[$unquotedTableName]);
+            }
+
+            $indexes = [];
+            if (isset($indexesByTable[$unquotedTableName])) {
+                $indexes = $this->_getPortableTableIndexesList($indexesByTable[$unquotedTableName], $tableName);
```

Comment:
Does `_getPortableTableIndexesList()` expect the quoted table name as the 2nd parameter? I cannot see it used anywhere in the implementation of `_getPortableTableIndexesList()` besides dispatching some event. Could you please explain what it is for here?

Reply:
Just passing over to honour the arguments and the chain or parent calls. The code here does not do anything with that.